### PR TITLE
Add ability to delete Pipelines, Tasks, and ClusterTasks

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Resolves https://github.com/tektoncd/dashboard/issues/775

Add the ability to delete Pipelines, Tasks, and ClusterTasks, both
individually and in bulk via the Carbon Table batch actions toolbar.

There's one slight difference in the user experience for these resource
types compared to the existing delete functionality for PipelineRuns
and TaskRuns. That is, the delete action for a single resource is
exposed as a button inline on the table row rather than being nested
in an overflow menu. This is because we only have at most 2 actions
on a row, one being the link to the details page for the resource.
Carbon guidelines recommend keeping the actions inline when there are
fewer than 3 actions.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/containers/PipelineRuns/PipelineRuns.js [hunk 1]
```diff
@@ -345,20 +348,22 @@ export /* istanbul ignore next */ class PipelineRuns extends Component {
             lowContrast
           />
         )}
-        {this.state.submitError && (
+        {this.state.deleteError && (
```
Comment (steveodonovan) [inline src/containers/PipelineRuns/PipelineRuns.js:351]: Not for this PR but just spotted that were missing error handling for cancelling a run :see_no_evil:

https://github.com/tektoncd/dashboard/blob/266f8030f92a3d321ba648b1d97512f8c2280dfe/src/containers/PipelineRuns/PipelineRuns.js#L115

**Review by steveodonovan — APPROVED** [review]
Discussed on follow up item in slack

/approve
/lgtm

File: src/containers/PipelineRuns/PipelineRuns.js (hunk 1 above)
Comment (AlanGreene) [reply src/containers/PipelineRuns/PipelineRuns.js:351]: so we are... created an issue to track that: https://github.com/tektoncd/dashboard/issues/1884

**Pull request merged**